### PR TITLE
dynamically pull the matrix from foreman.git

### PR DESCRIPTION
**Pull request by evgeni**
needs https://github.com/theforeman/foreman/pull/9865

### CODE REVIEW

File: .github/workflows/foreman_plugin.yml [hunk 1]
```diff
@@ -29,9 +29,13 @@ jobs:
     steps:
       - name: Build test matrix
         id: build_matrix
-        # TODO: Make this dynamic, based on the Foreman version
         run: |
-          echo 'matrix={"ruby": ["2.7"], "node": ["12"]}' >> "$GITHUB_OUTPUT"
+          {
+            echo 'matrix<<EOF'
+            #curl --silent --show-error --fail https://raw.githubusercontent.com/theforeman/foreman/${{ inputs.foreman_version }}/.github/matrix.json
+            curl --silent --show-error --fail https://raw.githubusercontent.com/evgeni/foreman/enter-the-matrix/.github/matrix.json
```
Comment (evgeni) [inline .github/workflows/foreman_plugin.yml]: this needs the correct URL, once [the PR](https://github.com/theforeman/foreman/pull/9865) is merged ;-)

Comment (ekohl) [reply .github/workflows/foreman_plugin.yml]: Merged that PR. We should cherry pick it though.

File: .github/workflows/foreman_plugin.yml [hunk 2]
```diff
@@ -29,9 +29,13 @@ jobs:
     steps:
       - name: Build test matrix
         id: build_matrix
-        # TODO: Make this dynamic, based on the Foreman version
         run: |
-          echo 'matrix={"ruby": ["2.7"], "node": ["12"]}' >> "$GITHUB_OUTPUT"
+          {
+            echo 'matrix<<EOF'
```
Comment (ekohl) [inline .github/workflows/foreman_plugin.yml:34]:
```suggestion
            echo 'matrix=<<EOF'
```

Comment (evgeni) [reply .github/workflows/foreman_plugin.yml:34]: I also do find the syntax odd, and yet it is correct:
https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#multiline-strings

(tested here: https://github.com/evgeni/gha-playground/actions/runs/6544062561)

File: .github/workflows/foreman_plugin.yml (hunk 1 above)
Comment (evgeni) [reply .github/workflows/foreman_plugin.yml]: does 3.6+ sound reasonable?

**Comment by evgeni** [comment]
FTR, run with the correct URL: https://github.com/evgeni/gha-playground/actions/runs/6546000881/job/17775580671

File: .github/workflows/foreman_plugin.yml (hunk 2 above)
Comment (ekohl) [reply .github/workflows/foreman_plugin.yml:34]: Ah yes, I remember going over this part of the docs but I've never used it before.

**Review by ekohl — APPROVED** [review]

File: .github/workflows/foreman_plugin.yml (hunk 1 above)
Comment (ekohl) [reply .github/workflows/foreman_plugin.yml]: Right now we don't use it for anything yet, so I'd say so. If any plugin starts using it and needs an older version, we can change it.

**Pull request merged**